### PR TITLE
pkg: replace Decoder with SeekTable

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
## Breaking API Changes
- Removed `Decoder` and `NewDecoder`.
- Added `NewSeekTable([]byte)` as the metadata-only seek-table API.
- Replaced decoder metadata lookups `GetIndexByID` and `GetIndexByDecompOffset` with `EntryByID` and `EntryByDecompressedOffset`, returning `(FrameOffsetEntry, bool)` values.
- Replaced decoder `Size() int64` with seek-table `Size() uint64`.
- Moved `FrameOffsetEntry`, `REnvironment`, and `WEnvironment` to the top-level package; `pkg/env` is no longer part of the public API.
- Changed `WithREnvironment` and `WithWEnvironment` to accept the top-level environment interfaces.

## Problem
- `NewDecoder` reused `readerImpl` even though metadata lookup does not need decompression, reader offsets, frame cache, or `Close` semantics.
- `NewDecoder` required an unused `ZSTDDecoder` and built a panic-only read environment just to parse seek-table bytes.
- The old metadata lookup API returned pointers into internal index storage, allowing callers to mutate parsed state.
- `env.FrameOffsetEntry` leaked implementation plumbing into normal reader and metadata APIs.
- Metadata size was signed even though seek-table offsets and sizes are unsigned.

## Solution
- Make parsed seek-table metadata the first-class API with `NewSeekTable` and a private concrete table type.
- Keep `Reader` focused on `io.Reader`, `io.ReaderAt`, `io.Seeker`, and `io.Closer`; callers use `NewSeekTable` for metadata inspection.
- Store parsed entries directly on `seekTable` and use `sort.Search` for decompressed-offset lookup plus direct indexing for ID lookup.
- Return `(FrameOffsetEntry, bool)` values from lookup methods to avoid nil checks and internal pointer exposure.
- Promote frame entries and custom environment hooks to top-level package types.

## Summary
- Separate metadata parsing from reader/decompression state.
- Remove the reader-shaped decoder shortcut and its close/panic-env edge cases.
- Add seek-table tests, fuzz coverage, and benchmarks for 16K, 128K, and 1M entries.

### CODE REVIEW

File: pkg/seek_table.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package seekable
+
+type seekTable struct {
```
[the remark this answers is not in the document]
Comment (SaveTheRbtz) [reply pkg/seek_table.go:5]: it is private so that users can't create it and hence can only obtain one via constructor
